### DISCUSSION TITLE
How to change color from params with Typescript?

### DISCUSSION THREAD

**Discussion by spencerchang** [heroui-inc/heroui#808] [Help]
I'm sure my params is one of color's string. 

EX1:
color="primary" -> ok

EX2: 
let colorPrimary:string = "primary"
color={colorPrimary} -> fail

(property) status?: "error" | "secondary" | "default" | "primary" | "success" | "warning" | undefined
Type 'string' is not assignable to type '"error" | "secondary" | "default" | "primary" | "success" | "warning" | undefined'.ts(2322)

<img width="727" alt="截圖 2022-10-09 下午5 43 56" src="https://user-images.githubusercontent.com/10862903/194749821-705c6137-2378-4644-8075-cafb47d1a8cb.png">

**Comment by tianenpang** [comment]
Hi @spencerchang you need to give `helper.color` a correct type that matches the `Input` [status color's type](https://nextui.org/docs/components/input#simple-colors).

For example:
```tsx
import { Input } from '@nextui-org/react';
import type { InputProps, SimpleColors } from '@nextui-org/react';

type Helper = {
  // one of the SimpleColors
  color?: SimpleColors;

  // one of the InputProps['color'] which is SimpleColors
  // color?: InputProps['color'];

  // default, error and undefined only
  // color?: 'default' | 'error';
};

const Index = () => {
  const helper: Helper = { color: 'default' };

  return <Input color={helper.color ?? 'primary'} />;
};
```
